src/Dialogs/Cloud/WeGlideUploadDialog.cpp

```diff
@@ -0,0 +1,191 @@
+    const WeGlide::Pilot& pilot,
+    uint_least32_t glider_id,
+    UploadResponse& response,
+    ProgressListener& progress) noexcept {
+    boost::json::value json =
+      co_await WeGlide::UploadFlight(*Net::curl, uri, pilot, glider_id,
+        igc_path, progress);
+    auto json_value = json.get_array().at(0);
+    response.scoring_date = UTF8ToWideConverter(
+      Json::Lookup(json_value, "scoring_date")->get_string().c_str());
+    response.flight_id = Json::Lookup(json_value, "id")->get_int64();
+    response.registration = UTF8ToWideConverter(
+      Json::Lookup(json_value, "registration")->get_string().c_str());
+    response.competition_id = UTF8ToWideConverter(
+      Json::Lookup(json_value, "competition_id")->get_string().c_str());
+    auto user = Json::Lookup(json_value, "user")->as_object();
+
+    response.pilot.id =
+      Json::Lookup(user, "id")->get_int64();
+    response.pilot.name = UTF8ToWideConverter(
+      Json::Lookup(user, "name")->get_string().c_str());
+
+    auto aircraft = Json::Lookup(json_value, "aircraft")->as_object();
+
+    response.glider.id = Json::Lookup(aircraft, "id")->get_int64();
+    response.glider.name = UTF8ToWideConverter(Json::Lookup(aircraft, "name")->get_string().c_str());
+    response.glider.kind = UTF8ToWideConverter(Json::Lookup(aircraft, "kind")->get_string().c_str());
+    response.glider.sc_class = UTF8ToWideConverter(Json::Lookup(aircraft, "sc_class")->get_string().c_str());
+
+  }
+
+  bool UploadFlightDialog(Path igc_path, WeGlide::Pilot pilot, uint_least32_t glider_id) {
+    const WeGlideSettings& weglide_settings =
+      CommonInterface::GetComputerSettings().weglide;
+    StaticString<0x1000> msg;
+    NarrowString<0x200> url(weglide_settings.default_url);
+    url += "/igcfile";
+
+    try {
+      if (glider_id == 0)
+        glider_id = CommonInterface::GetComputerSettings().plane
+        .weglide_glider_type;
+      if (pilot.id == 0) {
+        pilot = weglide_settings.pilot;
+      }
+      UploadResponse response;
+      PluggableOperationEnvironment env;
+      LogFormat(_("WeGlide Upload: %s"), igc_path.c_str());
+      if (!File::Exists(LocalPath(igc_path))) {
+        msg.Format(_("Error! File '%s' doesn't exist!"), igc_path.c_str());
+        LogFormat(_("WeGlide Upload: %s"), msg.c_str());
+        ShowMessageBox(msg, _T("WeGlide Upload"), MB_OK | MB_ICONEXCLAMATION);
+
+        return false;
+      }
+
+      if (ShowCoDialog(UIGlobals::GetMainWindow(), UIGlobals::GetDialogLook(),
+        _("Download"), UpdateTask(igc_path, url, pilot,
+          glider_id, response, env), &env) == false)
+      {
+        return false;
+      }
+
+      msg.Format(_("File upload '%s' was succesfull!\n\n"), igc_path.c_str());
+      msg.AppendFormat(_("%-20s: %u\n"), _("flight_id"), response.flight_id);
+      // Unfortunately on Linux you cannot use the NarrowString itself, you have
+      // to convert this with '.c_str()'
```

Comment:
You cannot ever pass NarrowString to a vararg function. That is not a Linux restriction.

Reply:
Ok, in my msvc build and in my android clang build it works like I expected - but I will removing thsi comment and use the .c_str() construct every time!